src/lib.rs

```diff
@@ -19,47 +19,68 @@ fn take_guess() -> char {
         .expect("More than one character was guessed.");
 }
 
-fn run_game_loop(config: &mut Config, input: char) -> GameState {
-    if !config.secret_word.contains(input) {
-        return GameState::LOST;
+struct PlayerState {
+    wrong_guesses: u32,
+}
+
+fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerState) -> GameState {
+    if !config.secret_word.contains(guess) {
+        player_state.wrong_guesses += 1;
+
+        if player_state.wrong_guesses > config.lives {
```

Comment:
[nit]: No strong opinions here. As an alternative to keeping track of both (1) the total number of lives and (2) the wrong guesses, you could also keep track of the players current lives and decrement it. This may also simplify the logic here slightly as you have to compute the remaining lives anyway on line 35.

What do you think?